Large update that improves package structure and documentation

chrisluedtke opened the pull request:
This merge represents months of tweaks. Mainly I've focused on package structure and usability for the current fiscal year. There remains a lot to do, such as documenting common use-cases and flagging site-specific code.

chrisluedtke committed:
minor updates

chrisluedtke committed:
added fancy_box_wait()

chrisluedtke committed:
large updates

chrisluedtke committed:
update

chrisluedtke committed:
fixed read-only bug due to SharePoint & Excel 2016

chrisluedtke committed:
update enrollment days minimum, clear old method, fix typo

chrisluedtke committed:
script to exit student sections

chrisluedtke committed:
fix typo

chrisluedtke committed:
refactor

chrisluedtke committed:
minor updates for new school year

chrisluedtke committed:
section enrollment sync tool

chrisluedtke committed:
SY19 updates

chrisluedtke committed:
consolidated scripts to simple_cysh

chrisluedtke committed:
remove obsolete notebooks

chrisluedtke committed:
remove more obsolete notebooks

chrisluedtke committed:
consolidate exit student_section scripts to simple_cysh

chrisluedtke committed:
ToT audit improved

chrisluedtke committed:
Update .gitignore

chrisluedtke committed:
relative salesforce credentials path

chrisluedtke committed:
added notes on a better way to do it, given privileges

chrisluedtke committed:
map out Chicago's scheduled scripts

chrisluedtke committed:
minor tweaks

chrisluedtke committed:
update with regex

chrisluedtke committed:
update language to module best practice

chrisluedtke committed:
updated to more modular design

chrisluedtke committed:
Update cyautomation/cyschoolhouse/tracker_management.py

chrisluedtke committed:
added automated email script

chrisluedtke committed:
changed all \ in paths to \\ just to be sure

chrisluedtke committed:
cleaned imports and added automated email

chrisluedtke committed:
Update cyautomation/CHI - Script Schedule.ipynb

chrisluedtke committed:
re-incorporate Chicago's service trackers script

chrisluedtke committed:
split out student-section scripts to separate module

chrisluedtke committed:
added Chicago-specific section scripts

chrisluedtke committed:
minor tweaks

chrisluedtke committed:
more minor tweaks

chrisluedtke committed:
reorganize for easier package sharing

chrisluedtke committed:
remove Chicago jupyter notebooks

chrisluedtke committed:
package structure and function fixes

chrisluedtke committed:
clean repo

chrisluedtke committed:
rough requirements

chrisluedtke committed:
update main readme

chrisluedtke commented:
Hey @mrklees, I don't have permissions to merge. Can you enable that functionality? Also open to code review.

mrklees commented:
@chrisluedtke I think the conflicts are currently preventing the merge.  Once you've resolved those, then since you're a collaborator I think you should be able to merge the pull request after that.

chrisluedtke committed:
split out config variables and update readme

chrisluedtke committed:
tweak how Chicago-specific code is implemented

chrisluedtke committed:
load school reference table

remove school reference table from loading within functions, and instead save its path in config.py and load it at the head of each module
